src/core/lib/resolver/server_address.h

```diff
@@ -32,44 +31,74 @@
 
 namespace grpc_core {
 
-//
-// ServerAddress
-//
-
-// A server address is a grpc_resolved_address with an associated set of
-// channel args.  Any args present here will be merged into the channel
-// args when a subchannel is created for this address.
-class ServerAddress {
+class ResolverAttributeMap {
```

Comment:
Lets split this into its own file (preferably its own build target too if possible), and let's also give it some unit tests.